### ISSUE TITLE
make install-api fails on MacOS

### ISSUE THREAD

**Issue by avalonx** [idris-lang/Idris2#1493]
from a fresh git clone of Idris2 Master or after doing make clean, I get the same error when trying to: make install-api after a successful make install.

Error is: line 22: 30246 Illegal instruction: 4  "$DIR/idris2_app/idris2.so" "$@"
make: *** [install-api] Error 132

**Comment by gallais** [comment]
Do you have `zsh` installed?

**Comment by avalonx** [comment]
Meaning uninstall my homebrew installed gnu coreutils (version of BASH) to default to the coreutils supplied by MacOS which uses the zsh version of bash since #1418 obsoletes the need for the former? Okay I’ll try that. Thanks! 

Meanwhile it also looks like there is a need to run make clean after doing make install, before running the make install-api (not documented in the instructions). It looks like doing that might have worked as it gave no error and ran through the full make process but I’ve run out of time to verify. I’ll update on that by tomorrow.

**Comment by avalonx** [comment]
Here's the update.
1) gnu coreutils still required (installed via 'brew coreutils) as 'make bootstrap SCHEME=chez' still won't work without it. So I'm not sure what #1418 fixed.

2) Doing a 'make clean' before 'make install-api' seems to install the api components without errors.

3) However, it strangely creates ANOTHER 'idris2-0.30' directory under the existing 'idris2-0.30' directory and installs the api components into that. Is that how it is intended?
tree looks like this:
Idris2
--bin
--idris2-0.30
----//other stuff ...
----idris2-0.30
--------//api components

4) I've not compiled and tested the LSP server against it yet to know if it works. I'll do so next and update.

**Comment by ska80** [comment]
> 1. gnu coreutils still required (installed via 'brew coreutils) as 'make bootstrap SCHEME=chez' still won't work without it. So I'm not sure what #1418 fixed.

Yes, that is correct.  The generated scheme file for bootstrapping still uses `realpath` from `coreutils`.  Once the next version of Idris is released the new generated scheme file won't use `realpath` anymore.

**Comment by ska80** [comment]
> 3\. However, it strangely creates ANOTHER 'idris2-0.30' directory under the existing 'idris2-0.30' directory and installs the api components into that. Is that how it is intended?
>     tree looks like this:
>     Idris2
>     --bin
>     --idris2-0.30
>     ----//other stuff ...
>     ----idris2-0.30
>     --------//api components

The nested `idris2-0.30` is the API package and has the same version as  Idris itself as it should be.

**Comment by avalonx** [comment]
I compiled the LSP server and from VS Code I got the following errors:

Idris 2 LSP Client Output:
[Info  - 2:35:54 PM] Connection to server got closed. Server will restart.
[Info  - 2:35:54 PM] Connection to server got closed. Server will restart.
[Info  - 2:35:55 PM] Connection to server got closed. Server will restart.
[Info  - 2:35:55 PM] Connection to server got closed. Server will restart.
[Error - 2:35:56 PM] Connection to server got closed. Server will not be restarted.

idris2-lsp (Extension):
The Idris 2 LSP Client server crashed 5 times in the last 3 minutes. The server will not be restarted.
